src/main/java/uk/ac/ebi/embl/api/service/WebinSampleRetrievalService.java

```diff
@@ -0,0 +1,59 @@
+package uk.ac.ebi.embl.api.service;
+
+import uk.ac.ebi.ena.webin.cli.service.SampleService;
+import uk.ac.ebi.ena.webin.cli.service.SampleXmlService;
+import uk.ac.ebi.ena.webin.cli.validator.reference.Sample;
+
+public class WebinSampleRetrievalService implements SampleRetrievalService {
+
+    private String webinAuthToken;
```

Comment:
- Suggestion is to remove setters and make variables private final to have an immutable object. Quote:

The advantage of immutable objects is that you know their data cannot change, so you don't have to worry about that. You can pass them around freely without having to remember whether a method you pass them to could change them in a way your code is not prepared to handle. That makes working with immutable data easier.